### PR TITLE
REALTIME Table add purge feature

### PR DESCRIPTION
**Pull request by Airliquide76**
feature : ADD realtime table to task generator in order to purge them properly.
Only work on "DONE" segements

A bit hard to run on integration tests as segment does not have download URL on clusterIngrationTest.

Open to discussion.

### CODE REVIEW

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java [hunk 1]
```diff
@@ -150,4 +162,46 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
...
+    List<SegmentZKMetadata> segmentsZKMetadata = _clusterInfoAccessor.getSegmentsZKMetadata(realtimeTableName);
+
+    Map<Integer, LLCSegmentName> latestLLCSegmentNameMap = new HashMap<>();
+    for (SegmentZKMetadata segmentZKMetadata : segmentsZKMetadata) {
+      CommonConstants.Segment.Realtime.Status status = segmentZKMetadata.getStatus();
+      if (status.isCompleted()) {
+        completedSegmentsZKMetadata.add(segmentZKMetadata);
+      }
+
+      // Skip UPLOADED segments that don't conform to the LLC segment name
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java:185]: I don't think we want to skip the uploaded segments even if the name doesn't conform to the LLC segment name

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java [hunk 2]
```diff
@@ -92,26 +99,31 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
       } else {
         tableMaxNumTasks = Integer.MAX_VALUE;
       }
-      List<SegmentZKMetadata> offlineSegmentsZKMetadata = _clusterInfoAccessor.getSegmentsZKMetadata(tableName);
+      List<SegmentZKMetadata> segmentsZKMetadata;
```
Comment (Jackie-Jiang) [inline pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java:102]: We can read all the segments ZK metadata here, and for real-time table, remove the segments that is not completed

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java (hunk 1 above)
Comment (GSharayu) [reply pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java:185]: +1 We want purge jobs to handle real time segments uploaded. cc @sajjad-moradi

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java (hunk 2 above)
Comment (Airliquide76) [reply pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java]: It was right. I've removed wat was on function to get segment right on on the if section

File: pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java (hunk 1 above)
Comment (Airliquide76) [reply pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/purge/PurgeTaskGenerator.java]: Yes remove this never used part !

**Review by Jackie-Jiang — APPROVED** [review]

**Pull request merged**